core/integration/module_test.go

```diff
@@ -4696,6 +4696,61 @@ func (m *Test) GetRelDepSource(ctx context.Context, src *dagger.Directory) (*dag
 	})
 }
 
+func (ModuleSuite) TestContextDirectoryGit(ctx context.Context, t *testctx.T) {
+	testOnMultipleVCS(t, func(ctx context.Context, t *testctx.T, tc vcsTestCase) {
+		for _, mod := range []string{"context-dir", "context-dir-user"} {
+			// for _, mod := range []string{"context-dir"} {
+			t.Run(mod, func(ctx context.Context, t *testctx.T) {
+				c := connect(ctx, t)
+				mountedSocket, cleanup := privateRepoSetup(c, t, tc)
+				defer cleanup()
+
+				// c := connect(ctx, t)
+				devEngine := devEngineContainerAsService(devEngineContainer(c))
+
+				modRef := testGitModuleRef(tc, mod)
+				modGen := goGitBase(t, c).
+					WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+					WithServiceBinding("dev-engine", devEngine).
+					WithEnvVariable("_EXPERIMENTAL_DAGGER_RUNNER_HOST", "tcp://dev-engine:1234").
+					WithWorkdir("/work").
+					With(mountedSocket)
```

Comment:
NOTE: this is *not* right. But for some reason, the git credentials created by `privateRepoSetup` aren't properly propagated in nested sessions.